Deployment architecture docs

nrichers opened the pull request:
## Internal Notes for Reviewers

This PR adds a new section for deployment options, a new topic for configuring Azure Private Link, and makes some minor terminology changes to remove trigger words.
 
**LIVE PREVIEW**:

- [Deployment options](https://docs-demo.vm.validmind.ai/pr_previews/nrichers/sc-6306/document-deployment-options-for-validmind/about/deployment/deployment-options.html)
   -  [Multi-tenant cloud](https://docs-demo.vm.validmind.ai/pr_previews/nrichers/sc-6306/document-deployment-options-for-validmind/about/deployment/multi-tenant-cloud.html)
   - [Virtual Private ValidMind (VPV)](https://docs-demo.vm.validmind.ai/pr_previews/nrichers/sc-6306/document-deployment-options-for-validmind/about/deployment/virtual-private-validmind.html)
- [Configure Azure Private Link*](https://docs-demo.vm.validmind.ai/pr_previews/nrichers/sc-6306/document-deployment-options-for-validmind/guide/configuration/configure-azure-private-link.html)

* Untested, need Cullen to review with me. 

<!--
PR instructions for release notes:

1. Pick at least one label:

- `internal` (skip Step 2, no release notes required)
- `highlight`
- `enhancement`
- `bug`
- `deprecation`
- `documentation`

2. In the next section, describe the changes so that an external user can understand them. Keep it simple and link to the docs with [Learn more ...](<relative-link>), if available.
-->

## External Release Notes

We expanded our documentation on available deployment options for ValidMind, providing more information for both multi-tentant cloud and single-tentant options. [Learn more ...](/about/deployment/deployment-options.qmd)

nrichers commented:
@cachafla @validbeck my apologies, I had not changed the title of the PR before requesting review from you — this PR includes the docs for the deployment architecture info we had talked about.

validbeck commented on site/guide/configuration/configure-azure-private-link.qmd:
```diff
@@ -0,0 +1,82 @@
+  - ../configure-aws-privatelink.html
+---
+
+To keep your network traffic private and minimizes its attack surface, configure Azure Private Link[^1] to establish a private connection between ValidMind and your company network.
```
> To keep your network traffic private and minimizes its attack surface,

I can't really tell what the second half of this sentence is getting at — is it "to minimize its attack surface"? What is "it" referring to in this case (antecedent unclear, or maybe my brain is just dumb)?

nrichers replied:
It's a broken sentence I had cribbed from the AWS PrivateLink topic. Should have been: "To keep your network traffic private and minimize its attack surface ..." — now fixed in both places!

validbeck reviewed (changes requested):
Ohh, so much good info!

Now down to the mean stuff:

- Virtual Private ValidMind (VPV) — Maybe we want top add this to the variables?
- I'm not a fan of the request a demo button. Maybe make that section into a callout instead?
<img width="1191" alt="Screenshot 2024-10-04 at 3 15 39 PM" src="https://github.com/user-attachments/assets/6aed95e7-566c-4f90-99f4-48f703edf892">
- The "What's next" in `Managing your organization` needs the new Azure page added to it (and the grid-columns adjusted to display 2 on each row)
<img width="1181" alt="Screenshot 2024-10-04 at 3 16 15 PM" src="https://github.com/user-attachments/assets/94cf4703-d416-4398-b5ae-9f6b2a59a1c8">

- For the Azure Steps 1c - e maybe we want some line breaks here?
<img width="866" alt="Screenshot 2024-10-04 at 3 22 42 PM" src="https://github.com/user-attachments/assets/feb7fc01-9f6e-4a03-a92b-3df7a534851b">

cachafla commented:
> @cachafla @validbeck my apologies, I had not changed the title of the PR before requesting review from you — this PR includes the docs for the deployment architecture info we had talked about.

Sorry for the delay, I'll start reviewing.

cachafla commented on site/about/deployment/deployment-options.qmd:
```diff
@@ -0,0 +1,70 @@
+
+We offer two deployment models:
+
+- **Multi-tenant cloud** — Multiple organizations (tenants) share infrastructure while keeping data isolated, providing cost-efficiency and scalability. VPC support allows secure network connections.
```
Multi-tenant cloud — Multiple organizations (tenants) share infrastructure while keeping data isolated, ensuring cost-efficiency and scalability. For secure connectivity, a private link can be established to ensure traffic stays within your network, avoiding the public internet.

nrichers replied:
Changed verbatim to your suggestion.

cachafla commented on site/about/deployment/deployment-options.qmd:
```diff
@@ -0,0 +1,70 @@
+
+- **Multi-tenant cloud** — Multiple organizations (tenants) share infrastructure while keeping data isolated, providing cost-efficiency and scalability. VPC support allows secure network connections.
+
+- **Virtual Private ValidMind (VPV)** — A single-tenant architecture where one organization uses a dedicated infrastructure instance, offering greater control, customization, and enhanced security. This option is ideal for high-compliance needs and also supports VPC for secure connectivity.
```
Virtual Private ValidMind (VPV) — A single-tenant architecture where one organization uses dedicated infrastructure, offering greater control, customization, and enhanced security. This option is ideal for high-compliance needs, and secure connectivity can also be established via a private link.

nrichers replied:
Changed verbatim to your suggestion.

cachafla commented on site/about/deployment/deployment-options.qmd:
```diff
@@ -0,0 +1,70 @@
+
+We ensure data security through strong data isolation, encryption, and role-based access controls.[^1] Personal identifiable information and customer data are not stored in model documentation. For more information, see our data privacy policy.[^2]
+
+## Secure access
```
Secure access is provided via private link, and VPCs are just how we isolate individual ValidMind deployments, so it doesn't make much sense in this sentence.

Something like this could work:

Deployments support secure access via private link, giving you full control over your resources without exposing traffic to the public internet. Integration options include:

	•	AWS PrivateLink
	•	Google Cloud Private Service Connect
	•	Azure Private Link

Additionally, all connections are secured with HTTPS/SSL to ensure encrypted communication.

nrichers replied:
Changed verbatim as suggested.

cachafla commented:
In general I recommend using "private link connection" whenever we say "VPC connection". A VPC is just the "place" where a deployment. For example, we should say "private link" in this diagram:

![image](https://github.com/user-attachments/assets/0d76b524-2497-48ac-8846-e450fa373ae5)

nrichers commented:
@validbeck thank you for reviewing! I believe I have addressed your comments. 

> **Virtual Private ValidMind (VPV) — Maybe we want top add this to the variables?

Yup, done! I did not variablify "multi-tenant cloud", as it's not a proper name.

> I'm not a fan of the request a demo button. Maybe make that section into a callout instead?**

Done in both places (you had done one already). 

> For the Azure Steps 1c - e maybe we want some line breaks here?

You'd added some it seems but the spacing is at odds with what's further down, see the "Before" screenshot. I tried spacing out all substeps but the topic just ends up a mess of whitespace. As a compromise, I put the different options you select on their own bullets and then removed some of the spacing again. This puts the topic more in line with the other private link content we have. 

| Before | After |
|--------|-------|
| <img width="832" alt="image" src="https://github.com/user-attachments/assets/29708795-a8fa-4a93-a743-f03c127b9c02"> | <img width="860" alt="image" src="https://github.com/user-attachments/assets/63ac549e-8b62-43fb-a07b-aa99cd761612"> |

validbeck commented:
> As a compromise, I put the different options you select on their own bullets and then removed some of the spacing again. 

Nice, that was going to be my alternative suggestion!

nrichers commented:
@cachafla I believe I have addressed your comments.

> In general I recommend using "private link connection" whenever we say "VPC connection"

Thank you! Changed to "private link" or "private link connection in all the deployment content, for both text and graphics.

validbeck commented:
The callout was still quite cosmetically inelegant... so...

<img width="1193" alt="Screenshot 2024-10-15 at 3 08 36 PM" src="https://github.com/user-attachments/assets/97b1ebae-3b09-4710-9a95-fee443cc3307">
